### PR TITLE
feat: add zilliz component

### PR DESCRIPTION
**Pull request by zidanehakim**
Because

- We want to integrate VDP with zilliz vector db
- We want more vector db component choice

This commit

- Add milvus component along with the basic tasks (vector search, upsert, ..)
- Use api key authentication

### CODE REVIEW

File: data/zilliz/v0/assets/zilliz.svg
Comment (chuang8511) [inline data/zilliz/v0/assets/zilliz.svg:1]: I will provide you with the correct icon.

File: data/zilliz/v0/batchUpsert.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+	req := e.client.R().SetBody(reqParams).SetResult(&resp)
+
+	res, err := req.Post(upsertPath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode() != 200 {
```
Comment (chuang8511) [inline data/zilliz/v0/batchUpsert.go:43]: 200? 
I thought Post should be 201.
Could you confirm it?

File: data/zilliz/v0/createCollection.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	req := e.client.R().SetBody(reqParams).SetResult(&resp)
+
+	res, err := req.Post(createCollectionPath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode() != 200 {
```
Comment (chuang8511) [inline data/zilliz/v0/createCollection.go:92]: Please help me confirm the status code when the action is post

File: data/zilliz/v0/dropPartition.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+	req := e.client.R().SetBody(reqParams).SetResult(&resp)
+
+	res, err := req.Post(dropPartitionPath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode() != 200 {
```
Comment (chuang8511) [inline data/zilliz/v0/dropPartition.go:56]: please also confirm delete one!

**Review by chuang8511 — COMMENTED** [review]
Hi, though post or delete can be 200 for status code, I still hope you can re-confirm it!

File: data/zilliz/v0/createCollection.go
Comment (chuang8511) [inline data/zilliz/v0/createCollection.go:1]: Please change file name to snake_case

**Review by chuang8511 — COMMENTED** [review]
I missed the file name. Please change the file name

File: data/zilliz/v0/createCollection.go
Comment (zidanehakim) [reply data/zilliz/v0/createCollection.go:1]: Okay

File: data/zilliz/v0/createCollection.go (hunk 2 above)
Comment (zidanehakim) [reply data/zilliz/v0/createCollection.go:92]: According to its documentation, code will always be 0 for success request, otherwise it indicates errors

**Review by GeorgeWilliamStrong — COMMENTED** [review]
From documentation perspective, LGTM!

**Comment by zidanehakim** [comment]
> Hi, though post or delete can be 200 for status code, I still hope you can re-confirm it!

Based on zilliz docs, every success request will have 0 status code, otherwise it will be considered fail. Same goes for milvus, difference is milvus use 200 status code instead

**Pull request merged**